bundles/AdminBundle/public/js/pimcore/helpers.js

```diff
@@ -3235,3 +3239,76 @@ pimcore.helpers.buildMenu = function(items) {
         items[i].menu = Ext.create('pimcore.menu.menu', items[i].menu);
     }
 };
+
+pimcore.helpers.buildMainNavigationMarkup = function(menu) {
+    // priority for main menu starts at 10
+    // leaving enough space for bundles etc.
+
+    let priority = 10;
+    Object.keys(menu).forEach(key => {
+        if(menu[key].priority === undefined) {
+            menu[key].priority = priority;
+            priority += 10;
+        }
+    });
+
+
+    // sorting must be done manually here.
+    let sortedMenu = {};
+    Object.keys(menu).sort(function(a, b){
+        // a and b are the keys like file, extras e.t.c
+        // we use the keys to get the menu object themselfes with the priorities
+        return pimcore.helpers.priorityCompare(menu[a], menu[b]);
+    }).forEach(function(key) {
+        // after sorting the keys we then build a new sorted menu
+        sortedMenu[key] = menu[key];
+    });
+
+    let dh = Ext.DomHelper;
+    let ul = Ext.get("pimcore_navigation_ul");
+    const menuPrefix = 'pimcore_menu_';
+
+    Object.keys(sortedMenu).forEach(key => {
+        // the notifications are excluded from this
+        if(!sortedMenu[key]['exclude']) {
+            let li = {
+                id: menuPrefix + key,
+                tag: 'li',
+                cls: 'pimcore_menu_item pimcore_menu_needs_children',
+                html: '<div id="menuitem-' + key + '-iconEl" data-ref="iconEl" class="x-menu-item-main-icon x-menu-item-icon pimcore_main_nav_icon_' + sortedMenu[key]['iconCls'] + '"></div>',
+                'data-menu-tooltip': t(sortedMenu[key]['label'])
+            }
+            if(sortedMenu[key]['style']) {
+                li.style = sortedMenu[key]['style'];
+            }
+            dh.append(ul, li)
+        }
+    })
```

Comment:
We don't really need the intermediate variable, do we?

```suggestion
    const dh = Ext.DomHelper;
    const ul = Ext.get("pimcore_navigation_ul");
    const menuPrefix = 'pimcore_menu_';
    
    // sorting must be done manually here.
    Object.keys(menu).sort(function(a, b){
        // a and b are the keys like file, extras e.t.c
        // we use the keys to get the menu object themselfes with the priorities
        return pimcore.helpers.priorityCompare(menu[a], menu[b]);
    }).forEach(key => {
        // the notifications are excluded from this
        if(!menu[key]['exclude']) {
            const li = {
                id: menuPrefix + key,
                tag: 'li',
                cls: 'pimcore_menu_item pimcore_menu_needs_children',
                html: '<div id="menuitem-' + key + '-iconEl" data-ref="iconEl" class="x-menu-item-main-icon x-menu-item-icon pimcore_main_nav_icon_' + menu[key]['iconCls'] + '"></div>',
                'data-menu-tooltip': t(menu[key]['label'])
            }
            if(menu[key]['style']) {
                li.style = menu[key]['style'];
            }
            dh.append(ul, li)
        }
    })
```

Reply:
Good catch! We indeed don't need it! Thanks!